linode/terraform-provider-linode
# Resolved createTempKeys failure when bucket is deleted

## 📝 Description

Fixed error in edge cases where `createTempKeys` returns an error during refreshing states of a `linode_object_storage_object` resource after a bucket has been deleted at the back-end.

## ✔️ How to Test
### Manual Testing

> ***NOTE:*** The following steps assume that you have this PR checked out locally.

1. Create an Object Storage Bucket via Cloud Manager.
2. In a terraform-provider-linode sandbox environment (e.g. dx-devenv), apply the following terraform configuration. *Make sure to set your token, the bucket name, and the region appropriately.*
```
terraform {
  required_providers {
    linode = {
      source  = "linode/linode"
      version = "2.6.0"
    }
  }
}

provider "linode" {
  token = <MY_LINODE_TOKEN>
  obj_use_temp_keys = true
}

resource "linode_object_storage_object" "object" {
    bucket  = <MY_BUCKET_NAME>
    cluster = <MY_BUCKET_REGION>
    key     = "my-object"

    content          = "This is the content of the Object..."
    content_type     = "text/plain"
    content_language = "en"
}
```
3. After applying the terraform configuration, manually delete the newly created object and bucket via Cloud Manager.
4. Run `terraform refresh` to refresh the Terraform state. There should no longer be a `Bucket not found` error.
